Sources/BraintreeThreeDSecure/BTThreeDSecureRequest.swift

```diff
@@ -49,6 +49,7 @@ import BraintreeCore
     /// Non-Mastercard cardholders will fallback to a normal 3DS flow.
     public var dataOnlyRequested: Bool = false
 
+    // NEXT_MAJOR_VERSION remove cardAddChallenge in favor of cardAddChallengeRequested
```

Comment:
I had initially but apparently it blows up cocoapods unless you allow warnings ([CI failure here](https://github.com/braintree/braintree_ios/actions/runs/6693608803/job/18185106276)) which I don't think we want merchants doing. 😞 It was removed here: https://github.com/braintree/braintree_ios/pull/1122/commits/04c8c699e3c0856df29df029303c7f3b82fb23cb

Reply:
😢 One of the cons for moving to Swift is that there is [no macro](https://stackoverflow.com/questions/31540446/how-to-silence-a-warning-in-swift) for silencing deprecation warnings in-line. We already saw this limitation a bit when trying to deprecate the `start()` method on PPCP ([PR](https://github.com/paypal/paypal-ios/pull/212)).

I guess we can only deprecate something if it isn't used anywhere in our source code, to avoid warnings to be triggered. There might be some (slightly wonky) workaround we can do in our implementation to get around this. 

We could map the `cardAddChallenge` (deprecated) setting to an internal property (non-deprecated) so that referencing it doesn't trigger a warning.

Reply:
Yeah, it's pretty interesting to me that these attributes are pretty limited overall without allowing warnings. It'd be nice to have a cleaner way to mark properties/methods to be removed in a future version that works with all package managers. What _does_ work is this - just tested locally and `pod lib lint --subspec=ThreeDSecure` passes:
```swift
    #if !COCOAPODS
    @available(*, deprecated, renamed: "cardAddChallengeRequested", message: "Use cardAddChallengeRequested. This property will be removed in our next major version.")
    #endif
```
Not sure how we feel about only notifying SPM + Carthage merchants. Curious to hear what y'all think.